src/diffusers/models/transformers/auraflow_transformer_2d.py

```diff
@@ -68,6 +68,20 @@ def __init__(
         self.height, self.width = height // patch_size, width // patch_size
         self.base_size = height // patch_size
 
+    def pe_selection_index_based_on_dim(self, h, w):
+        h_p, w_p = h // self.patch_size, w // self.patch_size
+        original_pe_indexes = torch.arange(self.pos_embed.shape[1])
+        h_max, w_max = int(self.pos_embed_max_size ** 0.5), int(self.pos_embed_max_size ** 0.5)
+        original_pe_indexes = original_pe_indexes.view(h_max, w_max)
+        starth =  h_max // 2 - h_p // 2
+        endh = starth + h_p
+        startw = w_max // 2 - w_p // 2
+        endw = startw + w_p
+        original_pe_indexes = original_pe_indexes[
```

Comment:
Possible to to group together LoCs and add some comments to explain what is happening, broadly? Since many of our users also read the codebase, this will be useful.

Reply:
Ok!